pkg/client/client.go

```diff
@@ -930,6 +931,29 @@ func (c *Client) WaitForDeploymentRollout(ctx context.Context, dep *appsv1.Deplo
 	return nil
 }
 
+func (c *Client) WaitForDeploymentDeletion(ctx context.Context, dep *appsv1.Deployment) error {
+	var lastErr error
+	if err := wait.Poll(time.Second, deploymentDeleteTimeout, func() (bool, error) {
+		d, err := c.kclient.AppsV1().Deployments(dep.GetNamespace()).Get(ctx, dep.GetName(), metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			klog.V(4).ErrorS(err, "WaitForDeploymentDeletion: failed to get Deployment")
+			return false, nil
+		}
+		if d.Status.AvailableReplicas != 0 {
+			lastErr = errors.Errorf("got %d available replicas",
+				d.Status.AvailableReplicas)
+			return false, nil
+		}
+		return true, nil
```

Comment:
It could be easier to read if we checked first that the returned error is "not found"?

```suggestion
		if !apierrors.IsNotFound(err) {
			return true, nil
		}
		
		if err != nil {
			lastErr = err
		} else {
			lastErr = errors.Errorf("got %d available replicas",
				d.Status.AvailableReplicas)
		}
		
		return false, nil
```

Reply:
no strong feeling for me. By the way my initial suggestion was buggy! I've modified it to be more in-lined with what you mentioned.

Reply:
Yes, I also noticed ;) This way is better I'll update